From 61233a2d6f294023637e00e7964ee5fc8e64ed4b Mon Sep 17 00:00:00 2001
From: Ofer Heifetz <oferh@marvell.com>
Date: Tue, 15 Mar 2016 11:22:05 +0200
Subject: [PATCH 267/538] ARM: mvebu: L2/PCIe deadlock workaround addition

Original commit included small mechanism to allow each sub-architecture to
override the memory type used for PCI I/O mappings, simply make all of them
mapped MT_UNCACHED instead of MT_DEVICE.

But instead a different commit was merged by Russell:
'commit 1c8c3cf0b523 ("ARM: 8060/1: mm: allow sub-architectures to
override PCI I/O memory type")'
This commit did not include the correct PCI I/O mapping change needed for
the Marvell Armada 375 and Armada 38x SOCs (Cortex-A9 CPU core).

For a description of the original patch, see 'commit 497a92308af8 ("ARM:
 mvebu: implement L2/PCIe deadlock workaround")'

Change-Id: I35d3997027f4cd613c8f98ea6fba900138765449
Signed-off-by: Ofer Heifetz <oferh@marvell.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/28264
Tested-by: Star_Automation <star@marvell.com>
Reviewed-by: Lior Amsalem <alior@marvell.com>
---
 arch/arm/mach-mvebu/coherency.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/arch/arm/mach-mvebu/coherency.c b/arch/arm/mach-mvebu/coherency.c
index 958eedd..b03057f 100644
--- a/arch/arm/mach-mvebu/coherency.c
+++ b/arch/arm/mach-mvebu/coherency.c
@@ -190,6 +190,8 @@ static void __init armada_375_380_coherency_init(struct device_node *np)
 	coherency_cpu_base = of_iomap(np, 0);
 	arch_ioremap_caller = armada_pcie_wa_ioremap_caller;
 
+	pci_ioremap_set_mem_type(MT_UNCACHED);
+
 	/*
 	 * We should switch the PL310 to I/O coherency mode only if
 	 * I/O coherency is actually enabled.
-- 
1.9.1

